exp/topology/desiredstate/upgrade_plan.go

```diff
@@ -404,3 +408,43 @@ func GetUpgradePlanFromClusterClassVersions(clusterClassVersions []string) func(
 		return simplifiedUpgradePlan, nil, nil
 	}
 }
+
+// GetUpgradePlanFromExtension returns an upgrade plan by calling the GenerateUpgradePlan runtime extension.
+func GetUpgradePlanFromExtension(runtimeClient runtimeclient.Client, cluster *clusterv1.Cluster, extensionName string) func(ctx context.Context, desiredVersion, currentControlPlaneVersion, currentMinWorkersVersion string) ([]string, []string, error) {
```

Comment:
Let's add some simple unit tests for this func please.

Here is an example on how to do this with the fake Runtime Client: https://github.com/kubernetes-sigs/cluster-api/blob/2a010f05b37e800ced27153e9a4c7b34e406dafd/internal/controllers/clusterclass/clusterclass_controller_test.go#L418